### PR TITLE
`Communication`: Improve sidebar user interface design

### PR DESCRIPTION
**Pull request by asliayk**
#### General

- [X] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [X] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [X] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/development-process.html#naming-conventions-for-github-pull-requests).

#### Client
- [X] I **strictly** followed the [client coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client/).
- [X] Following the [theming guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client-design/), I specified colors only in the theming variable files and checked that the changes look consistent in both the light and the dark theme.
- [X] I added multiple integration tests (Jest) related to the features (with a high test coverage), while following the [test guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client-tests/).
- [X] I added multiple screenshots/screencasts of my UI changes.
- [X] I translated all newly inserted strings into English and German.

### Motivation and Context

**1st issue:** Next to each channel section header in sidebar, there was a "+" icon for channel creation. Even in sections where users couldn’t create a channel (e.g., the "exam" section), the "+" button was still visible. This was unnecessary and added clutter. And when this was clicked, the user first had to view the browse channel screen and then click the button in that dialog again to create a channel. Having too many icons and the need to open two dialogs one after the other was not user-friendly.

**2nd issue:** As discussed in the communication subgroup, the section header should not be visible when there are no channels/conversations in that section.

**3rd issue:** Although the prefixes "exercise-", "lecture-", and "exam-" were included in the names of the exercise, lecture, and exam channels respectively, their presence was unnecessary since the channels were already grouped into their corresponding sections. This made the interface look complicated.

### Description

**1st issue:** 
- Immediately above the sidebar, there is now a single button that, when clicked, opens a dropdown menu containing four options: "Create Direct Chat," "Create Group Chat," "Create Channel" and "Browse Channels." 
- The plus button next to each section header for adding channels are removed.

**2nd issue**: If a section is empty (i.e., there are no channels or conversations), the section header is no longer visible.

**3rd issue**: The channel prefixes ("exercise-" "lecture-" and "exam-") do not appear in their respective sections; they are only visible in the favorite and hidden sections.

### Steps for Testing

Prerequisites:
- 1 Instructor

1. Log in to Artemis
2. Navigate to Communication section of a course
3. Notice that there is a new button above the sidebar. By clicking on this button and selecting respective options, create direct chat, group chat and new channel. Then click on browse channels to open the browse channels dialog.
4. Notice that in exercise, exam and lecture sections, the channels have no prefixes ("exercise-" "lecture-" and "exam-"), but they become visible if you hide/favorite them and they are in hidden/favorites section.
5. If there is no hidden channel, the hidden section should not be visible. If there exist hidden channels, make them visible by clicking on their 3-dot button to see the options and click on "Show". When there is no hidden channel anymore, notice that hidden section is not visible.

[](https://artemis-test1.artemis.cit.tum.de)
[](https://artemis-test2.artemis.cit.tum.de)
[](https://artemis-test3.artemis.cit.tum.de)
[](https://artemis-test4.artemis.cit.tum.de)
[](https://artemis-test5.artemis.cit.tum.de)
[](https://artemis-test6.artemis.cit.tum.de)
[](https://artemis-test9.artemis.cit.tum.de)

#### Code Review
- [x] Code Review 1
- [x] Code Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

### Test Coverage

| Class/File | Line Coverage | Confirmation (assert/expect) |
|------------|--------------:|---------------------:|
| course-conversations.component.ts | 89.59% | ✅ ❌ |
| channels-overview-dialog.component.ts | 92.45% | ✅ ❌ |
| conversation-settings.component.ts | 89.15% | ✅ ❌ |
| course-users-selector.component.ts | 91.75% | ✅ ❌ |
| accordion-add-options.component.ts | not found (deleted) | ❌ |
| sidebar-accordion.component.ts | 100% | ✅ |
| sidebar-card-item.component.ts | 100% | ✅ |
| sidebar-card-large.component.ts | 93.75% | ✅ ❌ |
| sidebar-card-medium.component.ts | 90% | ✅ ❌ |
| sidebar-card-small.component.ts | 95% | ✅ ❌ |
| sidebar-card.directive.ts | 100% | ✅ |
| sidebar-event.service.ts | 100% | ✅ |
| sidebar.component.ts | 90.54% | ✅ ❌ |
| sidebar.ts | not found (modified) | ❌ |

### Screenshots

**Updated Sidebar Design**

## Release Notes

- **New Features**
  - Introduced new event bindings for creating channels and chats, enhancing user interaction.
  - Added a property to manage communication state, improving the component's responsiveness.
  - Implemented new methods for opening channel creation dialogs and managing dropdown visibility, streamlining the user experience.
  - Enhanced sidebar components with new dropdowns for chat and channel options, providing additional communication functionalities.

- **Bug Fixes**
  - Removed unnecessary elements from the channels overview dialog, simplifying the user interface.

- **Refactor**
  - Streamlined channel management logic and user interaction methods for clarity and efficiency.
  - Updated sidebar components to improve event handling and user interaction.

- **Documentation**
  - Updated localization strings for improved clarity and consistency across chat and channel functionalities.

### CODE REVIEW

**Review by anian03 — CHANGES_REQUESTED** [review]
Just tested it, looks good in terms of functionality!

Only suggestions for visual improvements:
- Can we left-align the contents of the dropdown menu? I think it looks a bit strange having centered text.
- Maybe consider adding icons to the items in the dropdown menu as well so that the user immediately finds the correct actions :)

**Review by SimonEntholzer — CHANGES_REQUESTED** [review]
Generally nice changes, but I think there are still some smaller UI issues that can be improved:

In dark mode the buttons aren't readable:
image

It would be nice if the "view" button shows up, when hovering anywhere within a box. Atm there is quite a margin, where the button does not show (that's rather a nitpick tbh.)
image

Here it seems like the highlight of the close button of the modals is on initially, until the user clicks somewhere else. 
image

File: src/main/webapp/app/overview/course-conversations/course-conversations.component.html [hunk 1]
```diff
@@ -29,10 +29,14 @@
                     [itemSelected]="conversationSelected"
                     [courseId]="course.id"
                     [sidebarData]="sidebarData"
-                    (onPlusPressed)="onAccordionPlusButtonPressed($event)"
+                    (onCreateChannelPressed)="onCreateChannelPressed()"
+                    (onBrowsePressed)="onBrowseChannelPressed()"
+                    (onDirectChatPressed)="onCreateDirectChatPressed()"
+                    (onGroupChatPressed)="onCreateGroupChatPressed()"
```
Comment (JohannesWt) [inline src/main/webapp/app/overview/course-conversations/course-conversations.component.html:35]: why did you introduce extra methods for the consumption of OutputEvents it did not just call the already existing functions directly?

File: src/main/webapp/app/overview/course-conversations/course-conversations.component.ts [hunk 2]
```diff
@@ -114,7 +115,9 @@ export class CourseConversationsComponent implements OnInit, OnDestroy {
     faFilter = faFilter;
     faSearch = faSearch;
 
-    // MetisConversationService is created in course overview, so we can use it here
+    createChannelFn?: (channel: ChannelDTO) => Observable<never>;
+    channelActions$ = new Subject<ChannelAction>();
+
```
Comment (JohannesWt) [inline src/main/webapp/app/overview/course-conversations/course-conversations.component.ts:120]: Could we also implement this by using the signal based output Event Emitter?

File: src/main/webapp/app/shared/sidebar/sidebar.component.scss [hunk 3]
```diff
@@ -41,3 +41,65 @@
...
+.btn-primary,
+.dropdown-menu,
+.dropdown-item {
+    box-sizing: border-box;
+}
+
+.dropdown-container {
+    position: relative;
+    width: 100%;
+    display: block;
```
Comment (JohannesWt) [inline src/main/webapp/app/shared/sidebar/sidebar.component.scss:55]: you can also do this with the bootstrap class attributes directly in the html template

File: src/main/webapp/app/shared/sidebar/sidebar.component.scss [hunk 4]
```diff
@@ -41,3 +41,65 @@
...
+    font-size: 0.88em;
+}
+
+.dropdown-menu {
+    position: absolute;
+    top: 100%;
+    left: 0;
+    right: 0;
+    background-color: white;
+    border: 1px solid #ccc;
```
Comment (JohannesWt) [inline src/main/webapp/app/shared/sidebar/sidebar.component.scss:72]: You should use the color variables bg-secondary, etc. (or similar) to choose the right colors also when switching to darkmode. You can get the correct variable through the Chrome Dev tools by selection text for example or the background of e.g. the Sidebar itself. (I think the variables are at the bottom of the styles bar in the Dev tools)

File: src/main/webapp/app/shared/sidebar/sidebar.component.scss [hunk 5]
```diff
@@ -41,3 +41,65 @@
...
+    z-index: 1000;
+    padding: 0;
+    box-shadow: 0 2px 5px rgba(0, 0, 0, 0.15);
+    border-radius: 0 0 4px 4px;
+    display: block;
+}
+
+.dropdown-item {
+    width: 100%;
+    padding: 10px 10px;
```
Comment (JohannesWt) [inline src/main/webapp/app/shared/sidebar/sidebar.component.scss:82]: we should also use the boostrap standards here e.g. `p-1` in the HTML template that we follow a similar look everywhere

File: src/main/webapp/app/shared/sidebar/sidebar.component.scss [hunk 6]
```diff
@@ -41,3 +41,65 @@
...
+    border-bottom: 1px solid #ccc;
+    cursor: pointer;
```
Comment (JohannesWt) [inline src/main/webapp/app/shared/sidebar/sidebar.component.scss:84]: there is a global class `.clickable` to achieve the same effect

File: src/main/webapp/app/shared/sidebar/sidebar.component.scss [hunk 7]
```diff
@@ -41,3 +41,65 @@
...
+
+.dropdown-item:hover {
+    background-color: #f1f1f1;
+}
+
+.dropdown-menu .dropdown-item {
+    display: flex;
+    align-items: center;
+    justify-content: center;
+    font-size: 14px;
```
Comment (JohannesWt) [inline src/main/webapp/app/shared/sidebar/sidebar.component.scss:102]: we should not define specific font sizes to keep a consistent look throughout the App

File: src/main/webapp/app/shared/sidebar/sidebar.component.ts [hunk 8]
```diff
@@ -24,7 +24,10 @@ import { ExerciseFilterModalComponent } from 'app/shared/exercise-filter/exercis
 export class SidebarComponent implements OnDestroy, OnChanges, OnInit {
     @Output() onSelectConversation = new EventEmitter<number>();
     @Output() onUpdateSidebar = new EventEmitter<void>();
-    @Output() onPlusPressed = new EventEmitter<string>();
+    @Output() onDirectChatPressed = new EventEmitter<void>();
+    @Output() onGroupChatPressed = new EventEmitter<void>();
+    @Output() onBrowsePressed = new EventEmitter<void>();
+    @Output() onCreateChannelPressed = new EventEmitter<void>();
```
Comment (JohannesWt) [inline src/main/webapp/app/shared/sidebar/sidebar.component.ts:30]: same here could we also use the signal based output?

File: src/main/webapp/app/shared/sidebar/sidebar.component.ts [hunk 9]
```diff
@@ -33,6 +36,7 @@ export class SidebarComponent implements OnDestroy, OnChanges, OnInit {
     @Input() channelTypeIcon?: ChannelTypeIcons;
     @Input() collapseState: CollapseState;
     @Input() showFilter: boolean = false;
+    @Input() inCommunication: boolean = false;
```
Comment (JohannesWt) [inline src/main/webapp/app/shared/sidebar/sidebar.component.ts:40]: here same for the input

```suggestion
    inCommunication = input<boolean>(false);
```

File: src/test/javascript/spec/component/overview/course-conversations/course-conversations.component.spec.ts [hunk 10]
```diff
@@ -155,6 +147,57 @@ examples.forEach((activeConversation) => {
...
+                // Set a mock course with id 1
+                component.course = { id: 1 } as Course;
+            });
+
+            it('should open the group chat creation dialog', fakeAsync(() => {
+                mockModalRef.result = Promise.resolve(undefined);
+                const spy = jest.spyOn(modalService, 'open').mockReturnValue(mockModalRef as NgbModalRef);
+                component.onCreateGroupChatPressed();
+                tick();
+                expect(spy).toHaveBeenCalledWith(GroupChatCreateDialogComponent, expect.anything());
```
Comment (JohannesWt) [inline src/test/javascript/spec/component/overview/course-conversations/course-conversations.component.spec.ts:169]: Could we be more specific here?
```suggestion
                expect(spy).toHaveBeenCalledExactlyOnceWith(GroupChatCreateDialogComponent, expect.anything());
```

File: src/test/javascript/spec/component/overview/course-conversations/course-conversations.component.spec.ts [hunk 11]
```diff
@@ -155,6 +147,57 @@ examples.forEach((activeConversation) => {
...
+            }));
+
+            it('should open the one-to-one chat creation dialog', fakeAsync(() => {
+                const spy = jest.spyOn(modalService, 'open').mockReturnValue(mockModalRef as NgbModalRef);
+                component.onCreateDirectChatPressed();
+                tick();
+                expect(spy).toHaveBeenCalledWith(OneToOneChatCreateDialogComponent, expect.anything());
```
Comment (JohannesWt) [inline src/test/javascript/spec/component/overview/course-conversations/course-conversations.component.spec.ts:176]: Same here and in the next occurrences (calledExactlyOnce)

**Review by JohannesWt — CHANGES_REQUESTED** [review]
I have a few comments regarding the use of bootstrap rather than own css classes. If you have any questions, just hit me up anytime

**Review by janthoXO — CHANGES_REQUESTED** [review]
when searching for a person in the new chat pop up. and deleting the search text again, the suggestion still gets displayed. If the input field is empty the suggestion should disappear.
Not quite sure if its directly related to this PR, but otherwise please create a follow up issue

https://github.com/user-attachments/assets/26a78611-d1b2-45b8-9b33-0e28ebdae14f

**Review by janthoXO — CHANGES_REQUESTED** [review]
The selection for private or group chat should disappear when the user clicks somewhere else. Same applies for the channel selection

https://github.com/user-attachments/assets/2f83fafc-18c8-4df4-b109-13566a4129b6

**Comment by janthoXO** [comment]
and for the future please keep in mind to write descriptive commit messages ;)

**Comment by asliayk** [comment]
After the discussion in the communication subgroup, it was decided to use a single button with a dropdown containing four options. I have also updated the code based on your previous reviews and would appreciate your further reviews :)

**Review by JohannesWt — DISMISSED** [review]
Thanks for the changes - Code looks good

**Review by edkaya — DISMISSED** [review]
Tested on TS5, works as expected. Maybe one nitpick: If you click on 'create channel' or 'add channel', we see a dropdown menu, which can not be closed if you click outside of the dropdown. I think dropdown should be closed if you click anywhere on the page.

**Review by anian03 — CHANGES_REQUESTED** [review]
Thanks for the changes, it looks nicer now 👍🏻

One issue I noticed: If you disable DMs and Group chats in the course settings, the corresponding button in the sidebar for creating new chats is still there. Could you remove it when DMs are disabled? We can't create new DMs there

File: src/main/webapp/app/overview/course-conversations/dialogs/channels-overview-dialog/channel-item/channel-item.component.scss [hunk 12]
```diff
@@ -1,4 +1,5 @@
 .channel-item {
+    padding: 10px;
```
Comment (rabeatwork) [inline src/main/webapp/app/overview/course-conversations/dialogs/channels-overview-dialog/channel-item/channel-item.component.scss:2]: Could you use rem values? e.g. 0.5rem or 0.75rem.

File: src/main/webapp/app/shared/course-users-selector/course-users-selector.component.ts [hunk 13]
```diff
@@ -121,6 +121,13 @@ export class CourseUsersSelectorComponent implements ControlValueAccessor, OnIni
         this.onChange(this.selectedUsers);
     }
 
+    onInputChange(event: Event): void {
+        const value = (event.target as HTMLInputElement).value;
+        if (value.length < 3) {
```
Comment (rabeatwork) [inline src/main/webapp/app/shared/course-users-selector/course-users-selector.component.ts:126]: Can you add a comment why `< 3`?

File: src/main/webapp/app/shared/sidebar/sidebar-card-item/sidebar-card-item.component.html [hunk 14]
```diff
@@ -51,7 +51,7 @@
                     @if (sidebarItem.icon) {
                         <fa-icon [fixedWidth]="true" [icon]="sidebarItem.icon" />
                     }
-                    {{ sidebarItem.title }}</span
+                    {{ removeChannelPrefix(sidebarItem.title) }}</span
```
Comment (rabeatwork) [inline src/main/webapp/app/shared/sidebar/sidebar-card-item/sidebar-card-item.component.html:54]: Please avoid using function calls in the html template. `removeChannelPrefix` also sounds communication specific, which is why this should rather happen in the communication component beforehand and not within the generic `sidebar-card-item.component`.

File: src/main/webapp/app/shared/sidebar/sidebar-card-item/sidebar-card-item.component.ts [hunk 15]
```diff
@@ -9,4 +9,16 @@ import { SidebarCardElement, SidebarTypes } from 'app/types/sidebar';
 export class SidebarCardItemComponent {
     @Input() sidebarItem: SidebarCardElement;
     @Input() sidebarType?: SidebarTypes;
+    @Input() groupKey?: string;
+
+    removeChannelPrefix(name: string): string {
```
Comment (rabeatwork) [inline src/main/webapp/app/shared/sidebar/sidebar-card-item/sidebar-card-item.component.ts:14]: Could you adapt the communication data before passing it to the sidebar-card.component?

File: src/main/webapp/app/shared/sidebar/sidebar-card-item/sidebar-card-item.component.ts [hunk 16]
```diff
@@ -9,4 +9,16 @@ import { SidebarCardElement, SidebarTypes } from 'app/types/sidebar';
...
+        const prefixes = ['exercise-', 'lecture-', 'exam-'];
+        const channelTypes = ['exerciseChannels', 'lectureChannels', 'examChannels'];
+        for (const prefix of prefixes) {
+            if (name && name.startsWith(prefix) && channelTypes.includes(<string>this.groupKey)) {
```
Comment (rabeatwork) [inline src/main/webapp/app/shared/sidebar/sidebar-card-item/sidebar-card-item.component.ts:18]:
```suggestion
            if (name?.startsWith(prefix) && channelTypes.includes(<string>this.groupKey)) {
```

File: src/main/webapp/i18n/de/student-dashboard.json [hunk 17]
```diff
@@ -68,8 +68,12 @@
                 "exerciseChannels": "Aufgaben",
                 "lectureChannels": "Vorlesungen",
                 "examChannels": "Klausuren",
+                "addChannelChat": "Kanal/Chat hinzufügen",
```
Comment (rabeatwork) [inline src/main/webapp/i18n/de/student-dashboard.json:71]: I figure you don't need this anymore

File: src/main/webapp/i18n/en/student-dashboard.json [hunk 18]
```diff
@@ -68,8 +68,12 @@
                 "exerciseChannels": "Exercises",
                 "lectureChannels": "Lectures",
                 "examChannels": "Exams",
+                "addChannelChat": "Add Channel/Chat",
```
Comment (rabeatwork) [inline src/main/webapp/i18n/en/student-dashboard.json:71]: I figure you don't need this anymore

File: src/main/webapp/i18n/en/student-dashboard.json [hunk 19]
```diff
@@ -68,8 +68,12 @@
...
                 "createChannel": "Create channel",
                 "browseChannels": "Browse channels",
+                "createChat": "Create Chat",
```
Comment (rabeatwork) [inline src/main/webapp/i18n/en/student-dashboard.json:74]: I figure you don't need this anymore

File: src/main/webapp/i18n/de/student-dashboard.json [hunk 20]
```diff
@@ -68,8 +68,12 @@
...
                 "createChannel": "Kanal erstellen",
                 "browseChannels": "Kanäle durchsuchen",
+                "createChat": "Chat erstellen",
```
Comment (rabeatwork) [inline src/main/webapp/i18n/de/student-dashboard.json:74]: I figure you don't need this anymore

File: src/main/webapp/app/shared/sidebar/sidebar-card-item/sidebar-card-item.component.ts [hunk 21]
```diff
@@ -9,4 +9,16 @@ import { SidebarCardElement, SidebarTypes } from 'app/types/sidebar';
 export class SidebarCardItemComponent {
     @Input() sidebarItem: SidebarCardElement;
     @Input() sidebarType?: SidebarTypes;
+    @Input() groupKey?: string;
+
+    removeChannelPrefix(name: string): string {
+        const prefixes = ['exercise-', 'lecture-', 'exam-'];
+        const channelTypes = ['exerciseChannels', 'lectureChannels', 'examChannels'];
+        for (const prefix of prefixes) {
```
Comment (rabeatwork) [inline src/main/webapp/app/shared/sidebar/sidebar-card-item/sidebar-card-item.component.ts:17]: personal preference: `prefixes.forEach((prefix) => { ... })`

**Review by rabeatwork — CHANGES_REQUESTED** [review]

**Comment by FelberMartin** [comment]
Tested the functionality as described in the PR, all features work as expected 👍

**Review by FelberMartin — APPROVED** [review]
Tested the functionality as described in the PR, all features work as expected 👍

**Review by sachmii — APPROVED** [review]
Tested on TS4. Everything works just fine, I really like these changes! Gives you a better overview over each channel and the channel creation process itself.

**Review by cremertim — APPROVED** [review]
code lgtm, one minor thing can be adapted.We should prefer to use Input<>()

File: src/main/webapp/app/shared/sidebar/sidebar-card-item/sidebar-card-item.component.ts [hunk 22]
```diff
@@ -9,4 +9,5 @@ import { SidebarCardElement, SidebarTypes } from 'app/types/sidebar';
 export class SidebarCardItemComponent {
     @Input() sidebarItem: SidebarCardElement;
     @Input() sidebarType?: SidebarTypes;
+    @Input() groupKey?: string;
```
Comment (cremertim) [inline src/main/webapp/app/shared/sidebar/sidebar-card-item/sidebar-card-item.component.ts:12]: New client guidelines want us to use of input <String>()
```suggestion
    groupKey = input<string>()
```

File: src/main/webapp/app/shared/sidebar/sidebar-card-large/sidebar-card-large.component.ts [hunk 23]
```diff
@@ -13,6 +13,8 @@ export class SidebarCardLargeComponent {
     @Input({ required: true }) sidebarItem: SidebarCardElement;
     @Input() sidebarType?: SidebarTypes;
     @Input() itemSelected?: boolean;
+    /** Key used for grouping or categorizing sidebar items */
+    @Input() groupKey?: string;
```
Comment (cremertim) [inline src/main/webapp/app/shared/sidebar/sidebar-card-large/sidebar-card-large.component.ts:17]: see comment aboe
```suggestion
     groupKey = input<string>()
```

File: src/main/webapp/app/shared/sidebar/sidebar-card-medium/sidebar-card-medium.component.ts [hunk 24]
```diff
@@ -16,6 +16,8 @@ export class SidebarCardMediumComponent {
     @Input() sidebarType?: SidebarTypes;
     @Input() itemSelected?: boolean;
     @Output() pageChange = new EventEmitter<number>();
+    /** Key used for grouping or categorizing sidebar items */
+    @Input() groupKey?: string;
```
Comment (cremertim) [inline src/main/webapp/app/shared/sidebar/sidebar-card-medium/sidebar-card-medium.component.ts:20]: see comment above

File: src/main/webapp/app/shared/sidebar/sidebar-card-large/sidebar-card-large.component.html [hunk 25]
```diff
@@ -5,5 +5,6 @@
     (click)="emitStoreAndRefresh(sidebarItem.id)"
     [routerLinkActive]="'bg-selected border-selected'"
 >
-    <jhi-sidebar-card-item [sidebarType]="sidebarType" [sidebarItem]="sidebarItem" />
+    <!-- groupKey is used to determine the display format of channel names in different sidebar sections -->
+    <jhi-sidebar-card-item [groupKey]="groupKey" [sidebarType]="sidebarType" [sidebarItem]="sidebarItem" />
```
Comment (cremertim) [inline src/main/webapp/app/shared/sidebar/sidebar-card-large/sidebar-card-large.component.html:9]: need to be accessed with groupKey() after implementing signals
```suggestion
    <jhi-sidebar-card-item [groupKey]="groupKey()" [sidebarType]="sidebarType" [sidebarItem]="sidebarItem" />
```

File: src/main/webapp/app/shared/sidebar/sidebar-card-small/sidebar-card-small.component.ts [hunk 26]
```diff
@@ -16,6 +16,8 @@ export class SidebarCardSmallComponent {
     @Input({ required: true }) sidebarItem: SidebarCardElement;
     @Input() sidebarType?: SidebarTypes;
     @Input() itemSelected?: boolean;
+    /** Key used for grouping or categorizing sidebar items */
+    @Input() groupKey?: string;
```
Comment (cremertim) [inline src/main/webapp/app/shared/sidebar/sidebar-card-small/sidebar-card-small.component.ts:20]: comment above

**Review by PaRangger — APPROVED** [review]
Tested on TS3. Everything works and looks great. 👍

**Review by HawKhiem — APPROVED** [review]
Tested on TS3. Everything works as described! After unhide the announcement channel, the hidden section becomes invisible as described!
Screenshot 2024-10-18 185836
Screenshot 2024-10-18 185921
Screenshot 2024-10-18 185930

**Review by anian03 — APPROVED** [review]
Thanks for the quick fix, lgtm!

**Review by krusche — APPROVED** [review]

**Pull request merged**